src/context-selection/period-selector-bar-item/use-date-limit.js

```diff
@@ -1,48 +1,28 @@
 import { useMemo } from 'react'
 import {
-    getFixedPeriodsForTypeAndDateRange,
-    addFullPeriodTimeToDate,
-    removeFullPeriodTimeToDate,
     selectors,
     useDataSetId,
     useMetadata,
     formatJsDateToDateString,
     getCurrentDate,
     useClientServerDateUtils,
     useClientServerDate,
+    getCurrentPeriodByType,
+    createUpdateFullPeriodTimeToDate,
 } from '../../shared/index.js'
 
-export const computeDateLimit = ({ dataSetId, metadata, fromClientDate }) => {
-    const currentDate = fromClientDate(getCurrentDate())
-
-    if (!dataSetId) {
-        return currentDate.serverDate
-    }
-
-    const dataSet = selectors.getDataSetById(metadata, dataSetId)
-    const periodType = dataSet?.periodType
-    const openFuturePeriods = dataSet?.openFuturePeriods || 0
-
-    if (openFuturePeriods <= 0) {
-        return currentDate.serverDate
-    }
-
-    // will only get the current period
-    const [currentPeriod] = getFixedPeriodsForTypeAndDateRange({
-        periodType,
-        startDate: removeFullPeriodTimeToDate(
-            currentDate.serverDate,
-            periodType
-        ),
-        endDate: currentDate.serverDate,
-    })
-
-    let startDateLimit = new Date(currentPeriod.startDate)
-
-    for (let i = 0; i <= openFuturePeriods; ++i) {
-        startDateLimit = addFullPeriodTimeToDate(startDateLimit, periodType)
-    }
-    return startDateLimit
+export const computePeriodDateLimit = ({
+    periodType,
+    serverDate,
+    openFuturePeriods = 0,
+}) => {
+    const currentPeriod = getCurrentPeriodByType(periodType, serverDate)
```

Comment:
with this approach, I think we need to have additional logic in case there is no currentPeriod from getPeriods + filter as of 1 January (i.e. for period types other than the yearly ones you have handled)?

Like currently, the SIX_MONTHLY_APRIL periodType is failing because the periods produced by getPeriods as of right now (19 January 2023) are [1 Apr 2023 - 30 Sep 2023, 1 Oct 2023 - 31 Mar 2024] and none of these are "current" (startDate less than the currentDate)

I assume that I'm noticing this for SIX_MONTHLY_APRIL, but it would apply to any period types that don't start as of 1 January (so I guess also WEEKLY types, BI_WEEKLY, SIX_MONTHLY_NOV)?

I think we probably also need to think about how this will work around 1 January because our period calculations are getYearWithOffset based on client date (so that it will open up with the current year from the user's perspective)?

Maybe it would be sufficient if the filter periods is empty, to regenerate periods based on the previous year?